react/components/Portal.tsx

```diff
@@ -0,0 +1,28 @@
+import React, { useLayoutEffect, useEffect, useState } from 'react'
+import ReactDOM from 'react-dom'
+
+export type ContainerType = React.ReactInstance | (() => React.ReactInstance | null) | null
+
+interface Props {
+  container?: ContainerType
+}
+
+function getContainer(container?: ContainerType) {
+  container = typeof container === 'function' ? container() : container
+  return ReactDOM.findDOMNode(container) as Element | null
+}
+
+const useEnhancedEffect = typeof window !== 'undefined' ? useLayoutEffect : useEffect
+
+const Portal: React.FC<Props> = props => {
+  const { children, container } = props
+  const [mountNode, setMountNode] = useState<Element | null>(null)
+
+  useEnhancedEffect(() => {
+    setMountNode(getContainer(container) || document.body)
+  }, [container])
+
+  return mountNode ? ReactDOM.createPortal(children, mountNode) : mountNode
+}
+
+export default Portal
```

Comment:
Why not use our own `Portal` from `vtex.react-portal`?

Reply:
The `vtex.react-portal` creates two divs with some styles and it seems like it should be used with the overlay component that creates two more divs with other styles. This Portal that I made is just a portal from React with no extra divs and my `Overlay` is the `Backdrop` component that uses just one div too.